recipes/gobject-introspection/all/conanfile.py

```diff
@@ -1,129 +1,180 @@
-        with tools.environment_append(
-            VisualStudioBuildEnvironment(self).vars
-        ) if self._is_msvc else tools.no_op():
-            meson = self._configure_meson()
-            meson.install()
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-        tools.rmdir(os.path.join(self.package_folder, "share"))
-        for pdb_file in glob.glob(os.path.join(self.package_folder, "bin", "*.pdb")):
-            os.unlink(pdb_file)
+        copy(self, "COPYING", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        meson = Meson(self)
+        meson.install()
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rm(self, "*.pdb", self.package_folder, recursive=True)
+        fix_apple_shared_install_name(self)
 
     def package_info(self):
-        self.cpp_info.names["pkg_config"] = "gobject-introspection-1.0"
+        self.cpp_info.set_property("pkg_config_name", "gobject-introspection-1.0")
         self.cpp_info.libs = ["girepository-1.0"]
-        self.cpp_info.includedirs.append(
-            os.path.join("include", "gobject-introspection-1.0")
-        )
-
-        bin_path = os.path.join(self.package_folder, "bin")
-        self.output.info("Appending PATH env var with: {}".format(bin_path))
-        self.env_info.PATH.append(bin_path)
+        self.cpp_info.includedirs.append(os.path.join("include", "gobject-introspection-1.0"))
 
         exe_ext = ".exe" if self.settings.os == "Windows" else ""
 
         pkgconfig_variables = {
-            'datadir': '${prefix}/res',
-            'bindir': '${prefix}/bin',
-            'g_ir_scanner': '${bindir}/g-ir-scanner',
-            'g_ir_compiler': '${bindir}/g-ir-compiler%s' % exe_ext,
-            'g_ir_generate': '${bindir}/g-ir-generate%s' % exe_ext,
-            'gidatadir': '${datadir}/gobject-introspection-1.0',
-            'girdir': '${datadir}/gir-1.0',
-            'typelibdir': '${libdir}/girepository-1.0',
+            "datadir": "${prefix}/res",
+            "bindir": "${prefix}/bin",
+            "libdir": "${prefix}/lib",
+            "g_ir_scanner": "${bindir}/g-ir-scanner",
+            "g_ir_compiler": "${bindir}/g-ir-compiler%s" % exe_ext,
+            "g_ir_generate": "${bindir}/g-ir-generate%s" % exe_ext,
+            "gidatadir": "${datadir}/gobject-introspection-1.0",
+            "girdir": "${datadir}/gir-1.0",
+            "typelibdir": "${libdir}/girepository-1.0",
         }
         self.cpp_info.set_property(
             "pkg_config_custom_content",
-            "\n".join("%s=%s" % (key, value) for key,value in pkgconfig_variables.items()))
+            "\n".join(f"{key}={value}" for key, value in pkgconfig_variables.items()),
+        )
+
+        # TODO: remove in conan v2
+        bin_path = os.path.join(self.package_folder, "bin")
+        self.env_info.PATH.append(bin_path)
```

Comment:
```suggestion
        self.buildenv_info.append_path("GI_GIR_PATH", os.path.join(self.package_folder, "res", "gir-1.0"))

        # TODO: remove in conan v2
        bin_path = os.path.join(self.package_folder, "bin")
        self.env_info.PATH.append(bin_path)
        self.env_info.GI_GIR_PATH.append(os.path.join(self.package_folder, "res", "gir-1.0"))
```
`gobject-introspection` and all packages using it need to export `GI_GIR_PATH` with the path to the installed .gir files to ensure that they are found automatically.